node/pkg/processor/vaa_processor.go

```diff
@@ -0,0 +1,473 @@
+	"github.com/certusone/wormhole/node/pkg/supervisor"
+	"github.com/wormhole-foundation/wormhole/sdk/vaa"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/proto"
+)
+
+// VAAReactor is a reactor for performing consensus on v1 VAA messages. It sits on top of a reactor.Manager and implements
+// VAA specific logic like persistence, signed VAA broadcasts, accounting and governor.
+type VAAReactor struct {
+	// msgC is a channel of observed emitted messages
+	msgC <-chan *common.MessagePublication
+	// obsC is a channel of VAAs used to forward to the reactor manager
+	obsC chan *vaaObservation
+	// obsvReqSendC is a send-only channel of outbound re-observation requests to broadcast on p2p
+	obsvReqSendC chan<- *gossipv1.ObservationRequest
+	// signedInC is a channel of inbound signed VAA observations from p2p
+	signedInC <-chan *gossipv1.SignedVAAWithQuorum
+	// gossipSendC is a channel of outbound messages to broadcast on p2p
+	gossipSendC chan<- []byte
+
+	attestationEvents *reporter.AttestationEventReporter
+	logger            *zap.Logger
+	db                *db.Database
+
+	manager *reactor.Manager[*vaaObservation]
+
+	governor        *governor.ChainGovernor
+	acct            *accountant.Accountant
+	acctReadC       <-chan *common.MessagePublication
+	pythnetVaas     map[string]PythNetVaaEntry
+	pythnetVAAsLock sync.Mutex
+}
+
+// NewVAAReactor creates a new VAAReactor.
+func NewVAAReactor(
+	db *db.Database,
+	msgC <-chan *common.MessagePublication,
+	setC <-chan *common.GuardianSet,
+	gossipSendC chan<- []byte,
+	obsvC <-chan *gossipv1.SignedObservation,
+	obsvReqSendC chan<- *gossipv1.ObservationRequest,
+	signedInC <-chan *gossipv1.SignedVAAWithQuorum,
+	gk *ecdsa.PrivateKey,
+	gst *common.GuardianSetState,
+	attestationEvents *reporter.AttestationEventReporter,
+	g *governor.ChainGovernor,
+	acct *accountant.Accountant,
+	acctReadC <-chan *common.MessagePublication,
+) *VAAReactor {
+	// obsC is an internal buffer that is used to forward pre-filtered messages to the reactor manager.
+	// See VAAReactor.Run for filtering logic.
+	obsC := make(chan *vaaObservation, 10)
+
+	r := &VAAReactor{
+		obsvReqSendC:      obsvReqSendC,
+		signedInC:         signedInC,
+		msgC:              msgC,
+		obsC:              obsC,
+		gossipSendC:       gossipSendC,
+		db:                db,
+		attestationEvents: attestationEvents,
+		governor:          g,
+		acct:              acct,
+		acctReadC:         acctReadC,
+		pythnetVaas:       map[string]PythNetVaaEntry{},
+	}
+
+	manager := reactor.NewManager[*vaaObservation]("vaa", obsC, obsvC, setC, gst, reactor.Config{
+		RetransmitFrequency: 5 * time.Minute,
+		QuorumGracePeriod:   2 * time.Minute,
+		QuorumTimeout:       24 * time.Hour,
+		UnobservedTimeout:   24 * time.Hour,
+		Signer:              reactor.NewEcdsaKeySigner(gk),
+		NetworkAdapter:      reactor.NewChannelNetworkAdapter(gossipSendC),
+	}, r, r)
+	r.manager = manager
+
+	return r
+}
+
+func (p *VAAReactor) Run(ctx context.Context) error {
```

Comment:
What do you think about doing the processing of each event (big select statement below) in it's own goroutine?